client/extensions/woocommerce/app/products/product-update.js

```diff
@@ -77,8 +81,36 @@ class ProductUpdate extends React.Component {
 		// TODO: Remove the product we added here from the edit state.
 	}
 
+	// TODO: In v1, this deletes a product, as we don't have trash management.
+	// Once we have trashing management, we can introduce 'trash' instead.
 	onTrash = () => {
-		// TODO: Add action dispatch to trash this product.
+		const { translate, site, product, deleteProduct } = this.props;
+		const areYouSure = translate( 'Are you sure you want to delete %(product)s?', {
+			args: { product: product.name }
+		} );
+		accept( areYouSure, function( accepted ) {
+			if ( ! accepted ) {
+				return;
+			}
+			const successAction = () => {
+				debounce( () => {
+					page.redirect( getLink( '/store/products/:site/', site ) );
+				}, 1000 )();
+				return successNotice(
+					translate( '%(product)s successfully deleted.', {
```

Comment:
Hi! I've found a possible matching string that has already been translated **23** times:
`translate( 'Successfully deleted @%(user)s', { context: 'Success message after a user has been modified.'} )` *ES Score: 7*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/woocommerce-product-trash-ui)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).